### PR TITLE
feat: support associating WAF without creating

### LINKED ISSUE #500: Support for associating an exising WAF
Is it currently possible to associate an existing WAF ACL?

### PR DESCRIPTION
**Pull request by rikhilrai**
closes #500 

Example serverless configuration:
```yml
wafConfig:
  enabled: true
  arn: arn:aws:waf-regional:<region>:<account-id>:rule/<rule-id>
```

### CODE REVIEW

**Comment by rikhilrai** [comment]
Raising a draft PR for now. Will wait for comments/suggestions before adding/updating tests & README

File: src/index.js [hunk 1]
```diff
@@ -1434,10 +1434,23 @@ class ServerlessAppsyncPlugin {
...
+    if (wafConfig.arn) {
+      return{
+        [wafAssocLogicalId]: {
+          Type: 'AWS::WAFv2::WebACLAssociation',
+          Properties: {
+            ResourceArn: { 'Fn::GetAtt': [apiLogicalId, 'Arn'] },
+            WebACLArn: wafConfig.arn,
+          },
+        },
+      }
```
Comment (rikhilrai) [inline src/index.js:1449]:
```suggestion
      };
```

**Comment by rikhilrai** [comment]
@bboure 

Thanks. Added a test 👍 
Separate PR with https://github.com/sid88in/serverless-appsync-plugin/pull/504 base branch #504

**Comment by rikhilrai** [comment]
@bboure re https://github.com/sid88in/serverless-appsync-plugin/pull/504#issuecomment-1147071848, I've updated the README. Is there another doc I should update?

**Review by bboure — APPROVED** [review]
LGTM

thanks!

**Pull request merged**